app/code/Magento/Customer/Model/ResourceModel/Customer.php

```diff
@@ -129,6 +129,17 @@ protected function _beforeSave(\Magento\Framework\DataObject $customer)
 
         if (!$customer->getEmail()) {
             throw new ValidatorException(__('The customer email is missing. Enter and try again.'));
+        } else {
+            if(!preg_match("/^([a-z0-9,!\#\$%&'\*\+\/=\?\^_`\{\|\}~-]|[\x{00A0}"
```

Comment:
Extract pattern to `const`. The better if you find existing pattern and refer to it.